### PR TITLE
[wasm] Trigger all the AOT library tests when relevant paths have

### PR DESCRIPTION
**Pull request by radical**
.. changes. This is used to trigger aot tests for changed paths like
`src/mono/wasm/runtime`. In such a case, all the AOT library tests will
run, and not just the smoke tests.

### CODE REVIEW

**Comment by radical** [comment]
@akoeplinger @safern Currently, for wasm/AOT jobs on a PR, we only run the smoke tests. This PR intends to change that to run *all* the library AOT tests when certain paths that would likely affect the AOT jobs.
I might narrow down the path checks to trigger this a bit though. This would help to catch issues that can otherwise slip by when merging. Does this sound acceptable?

**Comment by lewing** [comment]
I assume the failure here is the caching issue?

**Comment by radical** [comment]
> I assume the failure here is the caching issue?

Yep

File: eng/pipelines/common/templates/wasm-library-tests.yml [hunk 1]
```diff
@@ -32,11 +32,18 @@ jobs:
         value: ${{ parameters.alwaysRun }}
       - name: allWasmContainsChange
         value: $[ dependencies.evaluate_paths.outputs['SetPathVars_allwasm.containsChange'] ]
+      - name: _tryRunSmokeOnlyArg
+        value: ${{ parameters.tryRunSmokeOnly }}
+      - name: _runSmokeOnlyToUse
```
Comment (akoeplinger) [inline eng/pipelines/common/templates/wasm-library-tests.yml:37]: I don't have a solution in mind but this feels like going a little too far, we now have `_tryRunSmokeOnlyArg`, `_runSmokeOnlyToUse`, `_runSmokeTestsOnlyArg` and `runSmokeOnlyArg` variables/parameters which all do subtly different things.

I'd love if we could come up with a way to simplify this.

**Pull request closed without merging**